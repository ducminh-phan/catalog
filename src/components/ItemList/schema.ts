import { makeRequired, makeValidate } from "mui-rff";
import * as yup from "yup";

const schema = yup.object().shape({
  name: yup.string().required().min(5),
  description: yup.string().required(),
  price: yup.number().required(),
});

export const validate = makeValidate(schema);

export const required = makeRequired(schema);

export type FormData = yup.InferType<typeof schema>;
