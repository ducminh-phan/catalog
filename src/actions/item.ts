import { createAsyncThunk } from "@reduxjs/toolkit";

import Request from "utils/api";
import * as types from "utils/types";

type FetchItemsRequest = types.PaginationParams & {
  categoryId?: string;
};

export type FetchItemsResponse = {
  totalItems: number;
  items: types.Item[];
};

export const fetchItems = createAsyncThunk(
  "items/fetch",
  ({ categoryId, offset, limit }: FetchItemsRequest) =>
    Request.get<FetchItemsResponse>(
      `categories/${categoryId}/items?offset=${offset}&limit=${limit}`,
    ),
);

type EditItemsRequest = {
  categoryId?: string;
  itemId: string;
  data: {
    name: string;
    description: string;
    price: number;
  };
};

type EditItemsResponse = types.Item;

export const editItem = createAsyncThunk(
  "items/edit",
  ({ categoryId, itemId, data }: EditItemsRequest) =>
    Request.put<EditItemsResponse>(
      `categories/${categoryId}/items/${itemId}`,
      data,
    ),
);

type AddItemRequest = {
  categoryId?: string;
  data: {
    name: string;
    description: string;
    price: number;
  };
};

type AddItemResponse = types.Item;

export const addItem = createAsyncThunk(
  "item/add",
  ({ categoryId, data }: AddItemRequest) =>
    Request.post<AddItemResponse>(`categories/${categoryId}/items`, data),
);

type DeleteItemRequest = {
  categoryId?: string;
  itemId: string;
};

export const deleteItem = createAsyncThunk(
  "item/delete",
  ({ categoryId, itemId }: DeleteItemRequest) =>
    Request.delete(`categories/${categoryId}/items/${itemId}`),
);
