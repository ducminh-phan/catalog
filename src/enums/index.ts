export const ITEMS_PER_PAGE = 2;

export const CATEGORIES_PER_PAGE = 2;
