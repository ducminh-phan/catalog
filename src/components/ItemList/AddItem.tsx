import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from "@material-ui/core";
import AddIcon from "@material-ui/icons/Add";
import { unwrapResult } from "@reduxjs/toolkit";
import { TextField } from "mui-rff";
import React, { ReactNode } from "react";
import { Form } from "react-final-form";
import { connect, ConnectedProps } from "react-redux";

import * as itemActions from "actions/item";
import { useNotification } from "contexts/notification";

import { FormData, required, validate } from "./schema";

const mapDispatchToProps = {
  addItem: itemActions.addItem,
};

const connector = connect(null, mapDispatchToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

type Props = PropsFromRedux & {
  categoryId?: string;
};

const AddItem = (props: Props): React.ReactElement => {
  const { categoryId, addItem } = props;
  const [open, setOpen] = React.useState(false);
  const { setNotification } = useNotification();

  const handleOpen = (): void => {
    setOpen(true);
  };

  const handleClose = (): void => {
    setOpen(false);
  };

  const handleAdd = async (data: FormData): Promise<void> => {
    const resultAction = await addItem({
      categoryId,
      data,
    });

    handleClose();

    try {
      unwrapResult(resultAction);
      setNotification({
        severity: "success",
        message: "Add item successfully",
      });
    } catch (e) {
      setNotification({
        severity: "error",
        message: e.message ?? "Add item failed",
      });
    }
  };

  return (
    <>
      <Button color="inherit" onClick={handleOpen}>
        <AddIcon />
      </Button>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="item-add-form-title"
        fullWidth
      >
        <DialogTitle id="item-add-form-title">Add Item</DialogTitle>
        <DialogContent>
          <Form
            onSubmit={handleAdd}
            validate={validate}
            render={({ handleSubmit, invalid }): ReactNode => (
              <form onSubmit={handleSubmit} noValidate>
                <TextField label="Name" name="name" required={required.name} />
                <TextField
                  label="Description"
                  name="description"
                  required={required.description}
                />
                <TextField
                  label="Price"
                  name="price"
                  type="number"
                  required={required.price}
                />
                <DialogActions>
                  <Button type="button" onClick={handleClose}>
                    Cancel
                  </Button>
                  <Button color="primary" type="submit" disabled={invalid}>
                    Submit
                  </Button>
                </DialogActions>
              </form>
            )}
          />
        </DialogContent>
      </Dialog>
    </>
  );
};

export default connector(AddItem);
