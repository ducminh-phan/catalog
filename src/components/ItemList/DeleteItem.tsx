import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from "@material-ui/core";
import DeleteIcon from "@material-ui/icons/Delete";
import { unwrapResult } from "@reduxjs/toolkit";
import React from "react";
import { connect, ConnectedProps } from "react-redux";

import * as itemActions from "actions/item";
import { useNotification } from "contexts/notification";

const mapDispatchToProps = {
  deleteItem: itemActions.deleteItem,
};

const connector = connect(null, mapDispatchToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

type Props = PropsFromRedux & {
  categoryId?: string;
  itemId: string;
};

const DeleteItem = (props: Props): React.ReactElement => {
  const { categoryId, itemId, deleteItem } = props;
  const [open, setOpen] = React.useState(false);
  const { setNotification } = useNotification();

  const handleOpen = (): void => {
    setOpen(true);
  };

  const handleClose = (): void => {
    setOpen(false);
  };

  const handleDelete = async (): Promise<void> => {
    const resultAction = await deleteItem({
      categoryId,
      itemId,
    });

    handleClose();

    try {
      unwrapResult(resultAction);
      setNotification({
        severity: "success",
        message: "Delete item successfully",
      });
    } catch (e) {
      setNotification({
        severity: "error",
        message: e.message ?? "Delete item failed",
      });
    }
  };

  return (
    <>
      <Button data-testid="delete-item" color="inherit" onClick={handleOpen}>
        <DeleteIcon />
      </Button>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="item-delete-form-title"
        fullWidth
      >
        <DialogTitle id="item-delete-form-title">Delete Item</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Are you sure to delete this item?
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button type="button" onClick={handleClose}>
            Cancel
          </Button>
          <Button color="primary" type="submit" onClick={handleDelete}>
            Delete
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

export default connector(DeleteItem);
