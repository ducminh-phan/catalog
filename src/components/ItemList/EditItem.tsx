import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from "@material-ui/core";
import EditIcon from "@material-ui/icons/Edit";
import { unwrapResult } from "@reduxjs/toolkit";
import { TextField } from "mui-rff";
import React, { ReactNode } from "react";
import { Form } from "react-final-form";
import { connect, ConnectedProps } from "react-redux";

import * as itemActions from "actions/item";
import { useNotification } from "contexts/notification";
import * as types from "utils/types";

import { FormData, required, validate } from "./schema";

const mapDispatchToProps = {
  editItem: itemActions.editItem,
};

const connector = connect(null, mapDispatchToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

type Props = PropsFromRedux & {
  categoryId?: string;
  item: types.Item;
};

const EditItem = (props: Props): React.ReactElement => {
  const { categoryId, editItem, item } = props;
  const [open, setOpen] = React.useState(false);
  const { setNotification } = useNotification();

  const handleOpen = (): void => {
    setOpen(true);
  };

  const handleClose = (): void => {
    setOpen(false);
  };

  const handleEdit = async (data: FormData): Promise<void> => {
    const resultAction = await editItem({
      categoryId,
      itemId: item.id.toString(),
      data,
    });

    handleClose();

    try {
      unwrapResult(resultAction);
      setNotification({
        severity: "success",
        message: "Edit item successfully",
      });
    } catch (e) {
      setNotification({
        severity: "error",
        message: e.message ?? "Edit item failed",
      });
    }
  };

  return (
    <>
      <Button data-testid="edit-item" color="inherit" onClick={handleOpen}>
        <EditIcon />
      </Button>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="item-edit-form-title"
        fullWidth
      >
        <DialogTitle id="item-edit-form-title">Edit Item</DialogTitle>
        <DialogContent>
          <Form
            onSubmit={handleEdit}
            initialValues={item}
            validate={validate}
            render={({ handleSubmit, invalid, dirty }): ReactNode => (
              <form onSubmit={handleSubmit} noValidate>
                <TextField
                  id="edit-item-name"
                  label="Name"
                  name="name"
                  required={required.name}
                />
                <TextField
                  id="edit-item-description"
                  label="Description"
                  name="description"
                  required={required.description}
                />
                <TextField
                  id="edit-item-price"
                  label="Price"
                  name="price"
                  type="number"
                  required={required.price}
                />
                <DialogActions>
                  <Button type="button" onClick={handleClose}>
                    Cancel
                  </Button>
                  <Button
                    color="primary"
                    type="submit"
                    disabled={invalid || !dirty}
                  >
                    Submit
                  </Button>
                </DialogActions>
              </form>
            )}
          />
        </DialogContent>
      </Dialog>
    </>
  );
};

export default connector(EditItem);
