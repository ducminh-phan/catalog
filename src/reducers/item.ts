import { createReducer } from "@reduxjs/toolkit";
import { normalize } from "normalizr";

import { editItem, fetchItems } from "actions/item";
import * as schemas from "utils/schemas";
import * as types from "utils/types";

interface ItemState {
  totalItems: number;
  items: {
    [key: string]: types.Item;
  };
  ids: string[];
}

const initialState: ItemState = {
  totalItems: 0,
  items: {},
  ids: [],
};

export default createReducer(initialState, (builder) =>
  builder
    .addCase(fetchItems.fulfilled, (state, action) => {
      const { payload } = action;
      const normalized = normalize(payload.items, schemas.items);

      return {
        totalItems: payload.totalItems,
        items: normalized.entities.items ?? {},
        ids: normalized.result,
      };
    })
    .addCase(editItem.fulfilled, (state, action) => {
      const { payload } = action;
      const normalized = normalize(payload, schemas.item);

      return {
        ...state,
        items: {
          ...state.items,
          ...normalized.entities.items,
        },
      };
    }),
);
