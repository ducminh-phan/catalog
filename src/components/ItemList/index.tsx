import {
  Box,
  List,
  ListItem,
  ListItemSecondaryAction,
  ListItemText,
  Paper,
} from "@material-ui/core";
import Pagination from "@material-ui/lab/Pagination";
import { RouteComponentProps } from "@reach/router";
import React, { ChangeEvent, ReactElement, useEffect, useState } from "react";
import { connect, ConnectedProps } from "react-redux";

import { fetchItems } from "actions/item";
import { useAuth } from "contexts/auth";
import { ITEMS_PER_PAGE } from "enums";
import { RootState } from "reducers";
import * as types from "utils/types";

import EditItem from "./EditItem";

interface ItemListProps extends RouteComponentProps {
  categoryId?: string;
}

const mapDispatchToProps = {
  fetchItems,
};

const connector = connect(
  ({ item }: RootState) => ({ item }),
  mapDispatchToProps,
);

type PropsFromRedux = ConnectedProps<typeof connector>;

type Props = PropsFromRedux & ItemListProps;

const ItemList = (props: Props): ReactElement => {
  // eslint-disable-next-line no-shadow
  const { fetchItems, item, categoryId } = props;

  const { data } = useAuth();
  const [offset, setOffset] = useState(0);
  const [page, setPage] = useState(1);

  useEffect(() => {
    fetchItems({ categoryId, offset, limit: ITEMS_PER_PAGE });
  }, [fetchItems, categoryId, offset]);

  const { totalItems, items } = item;
  const pageCount = Math.ceil(totalItems / ITEMS_PER_PAGE);

  const handleChangePage = (
    event: ChangeEvent<unknown>,
    value: number,
  ): void => {
    setPage(value);
    setOffset(ITEMS_PER_PAGE * (value - 1));
  };

  return (
    <Paper>
      <Box p={2}>
        <List>
          {Object.values(items).map((c: types.Item) => (
            <ListItem key={c.id}>
              <ListItemText primary={c.name} secondary={c.description} />
              {data?.user.id === c.userId ? (
                <ListItemSecondaryAction>
                  <EditItem categoryId={categoryId} item={c} />
                </ListItemSecondaryAction>
              ) : (
                <div />
              )}
            </ListItem>
          ))}
        </List>
        <Pagination count={pageCount} page={page} onChange={handleChangePage} />
      </Box>
    </Paper>
  );
};

export default connector(ItemList);
