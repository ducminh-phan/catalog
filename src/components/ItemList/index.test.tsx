import React from "react";

import { FetchItemsResponse } from "actions/item";
import { ITEMS_PER_PAGE } from "enums";
import Request from "utils/api";
import { fireEvent, render, setupUser, wait } from "utils/tests";

import ItemList from "./index";

const categoryId = "0";

it("renders the list of items correctly", async () => {
  jest.spyOn(Request, "get").mockImplementation(
    (): Promise<FetchItemsResponse> =>
      Promise.resolve({
        totalItems: 3,
        items: [1, 2].map((id) => ({
          id,
          userId: 0,
          name: `name ${id}`,
          description: `description ${id}`,
          price: 0,
        })),
      }),
  );

  // Render the first page
  const { baseElement, getByText, getAllByText } = render(
    <ItemList categoryId={categoryId} />,
  );
  await wait(() => getByText("name 1"));
  expect(baseElement).toMatchSnapshot();
  expect(getAllByText(/name.*/)).toHaveLength(2);
  expect(Request.get).toHaveBeenNthCalledWith(
    1,
    `categories/${categoryId}/items?offset=0&limit=${ITEMS_PER_PAGE}`,
  );

  // We can navigate to the next page
  const nextPageButton = getByText("2");
  expect(nextPageButton).toBeInTheDocument();

  jest.spyOn(Request, "get").mockImplementation(
    (): Promise<FetchItemsResponse> =>
      Promise.resolve({
        totalItems: 3,
        items: [
          {
            id: 3,
            userId: 0,
            name: `name 3`,
            description: `description 3`,
            price: 0,
          },
        ],
      }),
  );

  await wait(() => fireEvent.click(nextPageButton));
  expect(baseElement).toMatchSnapshot();
  expect(getAllByText(/name.*/)).toHaveLength(1);
  expect(Request.get).toHaveBeenLastCalledWith(
    `categories/${categoryId}/items?offset=${ITEMS_PER_PAGE}&limit=${ITEMS_PER_PAGE}`,
  );
});

it("show the button to add items to logged in users", async () => {
  setupUser();
  jest.spyOn(Request, "get").mockImplementation(
    (): Promise<FetchItemsResponse> =>
      Promise.resolve({
        totalItems: 3,
        items: [1, 2].map((id) => ({
          id,
          userId: 0,
          name: `name ${id}`,
          description: `description ${id}`,
          price: 0,
        })),
      }),
  );

  // Render the first page
  const { getByTestId, getByText } = render(
    <ItemList categoryId={categoryId} />,
  );
  await wait(() => getByText("name 1"));
  expect(getByTestId("add-item")).toBeInTheDocument();
});
