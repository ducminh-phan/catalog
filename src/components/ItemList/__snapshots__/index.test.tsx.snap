// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the list of items correctly 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root"
      >
        <ul
          class="MuiList-root MuiList-padding"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                name 1
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                description 1
              </p>
            </div>
            <div />
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                name 2
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                description 2
              </p>
            </div>
            <div />
          </li>
        </ul>
        <ul
          class="MuiList-root MuiList-padding"
        >
          <li
            class="MuiListItem-container"
          >
            <div
              class="MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction"
            >
              <nav
                aria-label="pagination navigation"
                class="MuiPagination-root"
              >
                <ul
                  class="MuiPagination-ul"
                >
                  <li>
                    <button
                      aria-label="Go to previous page"
                      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-disabled MuiButtonBase-disabled"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiPaginationItem-icon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                        />
                      </svg>
                    </button>
                  </li>
                  <li>
                    <button
                      aria-current="true"
                      aria-label="page 1"
                      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-selected"
                      tabindex="0"
                      type="button"
                    >
                      1
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </li>
                  <li>
                    <button
                      aria-label="Go to page 2"
                      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                      tabindex="0"
                      type="button"
                    >
                      2
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </li>
                  <li>
                    <button
                      aria-label="Go to next page"
                      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiPaginationItem-icon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
            <div
              class="MuiListItemSecondaryAction-root"
            >
              <div />
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders the list of items correctly 2`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root"
      >
        <ul
          class="MuiList-root MuiList-padding"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                name 3
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                description 3
              </p>
            </div>
            <div />
          </li>
        </ul>
        <ul
          class="MuiList-root MuiList-padding"
        >
          <li
            class="MuiListItem-container"
          >
            <div
              class="MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction"
            >
              <nav
                aria-label="pagination navigation"
                class="MuiPagination-root"
              >
                <ul
                  class="MuiPagination-ul"
                >
                  <li>
                    <button
                      aria-label="Go to previous page"
                      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiPaginationItem-icon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </li>
                  <li>
                    <button
                      aria-label="Go to page 1"
                      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
                      tabindex="0"
                      type="button"
                    >
                      1
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </li>
                  <li>
                    <button
                      aria-current="true"
                      aria-label="page 2"
                      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-selected"
                      tabindex="0"
                      type="button"
                    >
                      2
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </li>
                  <li>
                    <button
                      aria-label="Go to next page"
                      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-disabled MuiButtonBase-disabled"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiPaginationItem-icon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
            <div
              class="MuiListItemSecondaryAction-root"
            >
              <div />
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;
